src/components/application_manager/test/commands/mobile/change_registration_test.cc

```diff
@@ -0,0 +1,627 @@
+const std::string kVrSynonyms1 = "vr_synonyms_1";
+const std::string kVrSynonyms2 = "vr_synonyms_2";
+}  // namespace
+
+class ChangeRegistrationRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  ChangeRegistrationRequestTest()
+      : applications_(application_set_, applications_lock_) {}
+
+  void CreateDefaultMessage(MessageSharedPtr message) {
+    (*message)[am::strings::msg_params][am::strings::app_name] = kAppName1;
+    (*message)[am::strings::params][am::strings::connection_key] =
+        kConnectionKey;
+    (*message)[am::strings::msg_params][am::strings::vr_synonyms][0] =
+        kVrSynonyms1;
+    (*message)[am::strings::msg_params][am::strings::hmi_display_language] = 1;
+  }
+
+ protected:
+  void SetUp() OVERRIDE {
+    message_ = CreateMessage();
+    command_ = CreateCommand<ChangeRegistrationRequest>(message_);
+    app_ = CreateMockApp();
+  }
+
+  ChangeRegistrationRequestPtr command_;
+  MessageSharedPtr message_;
+  MockAppPtr app_;
+
+  am::ApplicationSet application_set_;
+  sync_primitives::Lock applications_lock_;
+  DataAccessor<am::ApplicationSet> applications_;
+  MockHMICapabilities mock_hmi_capabilities_;
+  NiceMock<policy_test::MockPolicyHandlerInterface> policy_interface_;
+};
+
+class ChangeRegistrationResponseTest
+    : public CommandsTest<CommandsTestMocks::kIsNice> {};
+
+TEST_F(ChangeRegistrationRequestTest, OnEvent_UnknownEvent_UNSUCCESS) {
+  Event event(hmi_apis::FunctionID::INVALID_ENUM);
+  EXPECT_CALL(mock_app_manager_, application(_)).Times(0);
+  command_->on_event(event);
+}
+
+TEST_F(ChangeRegistrationRequestTest, OnEvent_InvalidApp_UNSUCCESS) {
+  std::vector<hmi_apis::FunctionID::eType> function_id_vector;
+  function_id_vector.push_back(hmi_apis::FunctionID::UI_ChangeRegistration);
+  function_id_vector.push_back(hmi_apis::FunctionID::VR_ChangeRegistration);
+  function_id_vector.push_back(hmi_apis::FunctionID::TTS_ChangeRegistration);
+
+  (*message_)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
+
+  MockAppPtr invalid_app;
+  std::vector<hmi_apis::FunctionID::eType>::iterator it =
+      function_id_vector.begin();
+  for (; it != function_id_vector.end(); ++it) {
+    MessageSharedPtr event_message(CreateMessage());
+    (*event_message)[am::strings::params][am::hmi_response::code] = 1;
+    (*event_message)[am::strings::msg_params] = 0;
+    Event event(*it);
+    event.set_smart_object(*event_message);
+    EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+        .WillOnce(Return(invalid_app));
+    command_->on_event(event);
+  }
+}
+
+TEST_F(ChangeRegistrationRequestTest, OnEvent_ValidApp_SUCCESS) {
+  std::vector<hmi_apis::FunctionID::eType> function_id_vector;
+  function_id_vector.push_back(hmi_apis::FunctionID::UI_ChangeRegistration);
+  function_id_vector.push_back(hmi_apis::FunctionID::VR_ChangeRegistration);
+  function_id_vector.push_back(hmi_apis::FunctionID::TTS_ChangeRegistration);
+
```

Comment:
@Kozoriz I see this code for second time (function_id_vector, then push_backs), please use some function/method.